File: examples/python/advanced/error-handling/serdes-error.py
from aws_durable_execution_sdk_python import SerDesError

@durable_step
def process_complex_data(step_context: StepContext, data: object) -> dict:
    """Process data that might not be serializable."""
    try:
        # Process data
        return {"result": data}
    except SerDesError as e:
        # Handle serialization failure
        return {"error": "Cannot serialize result"}

File: examples/java/advanced/error-handling/serdes-error.java
try {
    var result = ctx.step("process-data", Map.class, stepCtx -> {
        // Process data that might not be serializable
        return Map.of("result", data);
    });
} catch (SerDesException e) {
    // Handle serialization failure
    System.out.println("Cannot serialize result: " + e.getMessage());
}
